lib/epics/ini.rb

```diff
@@ -23,9 +23,20 @@ def body
   end
 
   def key_signature
+    x_509_certificate_a = client.x_509_certificate_a
+
     Nokogiri::XML::Builder.new do |xml|
       xml.SignaturePubKeyOrderData('xmlns:ds' => 'http://www.w3.org/2000/09/xmldsig#', 'xmlns' => 'http://www.ebics.org/S001') {
         xml.SignaturePubKeyInfo {
+          if x_509_certificate_a
```

Comment:
The x509 xml data looks the same in all 3 places, just using a different input.

I would suggest moving that into a function/method to re-use and shorten the part of the xml document.